File: C++/73. Set Matrix Zeroes.cpp
class Solution {
public:
    void setZeroes(vector<vector<int>>& m) {
        int i,j;
        vector<int>r(m.size(),1);
        vector<int>c(m[0].size(),1);
        for(i=0;i<m.size();i++){
            for(j=0;j<m[0].size();j++){
                if(m[i][j]==0){
                    r[i]=0;
                    c[j]=0;
                }
            }
        }
        for(i=0;i<m.size();i++){
            for(j=0;j<m[0].size();j++){
                if(r[i]==0 || c[j]==0)
                m[i][j]=0;
            }
        }
    }
};

File: JAVA/73. Set Matrix Zeroes.java
class Solution {
    public void setZeroes(int[][] m) {
        int cs = m.length;
        int rs = m[0].length;
        boolean[] r = new boolean[cs];
        boolean[] c = new boolean[rs];
        
        for (int i = 0; i < cs; i++) {
            for (int j = 0; j < rs; j++) {
                if (m[i][j] == 0) {
                    r[i] = true;
                    c[j] = true;
                }
            }
        }
        
        for (int i = 0; i < cs; i++) {
            for (int j = 0; j < rs; j++) {
                if (r[i] || c[j]) {
                    m[i][j] = 0;
                }
            }
        }
    }
}
